src/lug/System/Win32/PathImplWin32.cpp

```diff
@@ -0,0 +1,91 @@
+#include <lug/System/Win32/PathImplWin32.hpp>
+
+#include <Shlobj.h>
+#include <utility>
+#include <windows.h>
+
+/*
+    Return Win32 root
+    Example: C:
+*/
+std::string lug::System::Path::priv::root() {
+    char* driveLetter = getEnv("HOMEDRIVE");
+    if (driveLetter == nullptr) {
+        return std::string("");
+    }
+
+    const std::string rootPath(driveLetter);
+
+    free(driveLetter);
+
+    return std::move(rootPath);
+}
+
+/*
+    Return Win32 home
+    Example: C:\Users\{USERNAME}
+*/
+std::string lug::System::Path::priv::home() {
+    char* driveLetter = getEnv("HOMEDRIVE");
+    char* homeDirectory = getEnv("HOMEPATH");
+
+    if (driveLetter == nullptr || homeDirectory == nullptr) {
+        free(driveLetter);
+        free(homeDirectory);
+        return std::string("");
+    }
+
+    std::string homePath(driveLetter);
+    homePath += homeDirectory;
+
+    free(driveLetter);
+    free(homeDirectory);
+
+    return std::move(homePath);
+}
+
+/*
+    Return the current working directory
+    Example:  D:\Vulkan\Lugdunum\Test
+*/
+std::string lug::System::Path::priv::cwd() {
+    TCHAR cwd[MAX_PATH];
+    if (FAILED(GetCurrentDirectory(MAX_PATH, cwd))) {
+        return std::string("");
+    }
+
+    std::wstring cwdString(&cwd[0]);
+    std::string cwdPath(cwdString.begin(), cwdString.end());

```

Comment:
That's not working like that. You can't convert `wstring` to `string` like that.